@here/harp-geometry/lib/SubdivisionModifier.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+    Face3,
+    Geometry,
+} from "three";
+
+/**
+ * The `SubdivisionModifier` subdivides triangle mesh geometries positioned
+ * on the surface of a sphere centered at `(0, 0, 0)`.
+ */
+export class SubdivisionModifier {
+    /**
+     * Constructs a new `SubdivisionModifier`.
+     *
+     * @param angle The maximum angle in radians between two vertices and the origin.
+     */
+    constructor(readonly angle: number) { }
+
+    /**
+     * Subdivides the faces of the given `THREE.Geometry`.
+     *
+     * @param geometry The `THREE.Geometry` to subdivide.
+     */
+    modify(geometry: Geometry): Geometry {
+        const { vertices, faces } = geometry;
+
+        const cache = new Map<string, number>();
```

Comment:
What is this cache caching?

Reply:
Middle vertices. 
`middleVertexCache` ?

Reply:
I added some documentation explaining what is that we are caching,